src/main/java/iterator/RichIterator.java

```diff
@@ -236,7 +333,24 @@ default String mkString() {
      * RichIterator.apply(1,2,3,4).append(5) // 1,2,3,4,5
      */
     default RichIterator<A> append(A elem) {
-        throw new NotImplementedException();
+        RichIterator<A> t = this;
+        return new RichIterator<A>() {
+            int i = 0;
```

Comment:
why `int` and not a `boolean`.
Theoretically speaking, this can "overflow" and reach zero value more than once.